pkg/gateway/utils_test.go

```diff
@@ -88,22 +143,61 @@ func Test_getAttachedPolicy(t *testing.T) {
 				},
 				policy: &anv1alpha1.TargetGroupPolicy{},
 			},
-			expectedK8sClientReturnedPolicy: targetGroupPolicyHappyPath,
-			want:                            targetGroupPolicyHappyPath,
-			expectPolicyNotFound:            false,
+			expectedK8sClientReturnedPolicies: []core.Policy{targetGroupPolicyHappyPath},
+			want:                              []core.Policy{targetGroupPolicyHappyPath},
+			expectPolicyNotFound:              false,
+		},
+		{
+			name: "Get gateway attached IAMAuthPolicy, happy path",
```

Comment:
What is the expected result of happy path? Typically, we shouldn't put "happy path" in the name of a test, but rather state what it is we expect the outcome to be